ecr-login/ecr.go

```diff
@@ -26,7 +27,20 @@ import (
 var notImplemented = errors.New("not implemented")
 
 type ECRHelper struct {
-	ClientFactory api.ClientFactory
+	clientFactory api.ClientFactory
+	logger        *logrus.Logger
+}
+
+func NewECRHelper(cf api.ClientFactory) ECRHelper {
+	return ECRHelper{
+		clientFactory: cf,
+		logger:        logrus.StandardLogger(),
+	}
+}
+
+func (self ECRHelper) WithLogOutput(w io.Writer) ECRHelper {
```

Comment:
I'm inclined to use FieldLogger here as well. The caller may want to customize the logger (e.g. use JSON logging)

Reply:
This would tie ecr-login to logrus as its logging implementation via its exported API. I'm not sure how important that flexibility is, but currently, and with this change, ecr-login could change to another logging library under the hood, without callers being aware or needing to make any changes.

AFAIK, if this only supported `WithFieldLogger`, a caller that wanted to disable ecr-login's logging entirely would have to implement a discarding `FieldLogger` implementation, or else take a dependency on logrus and call `logrus.New().SetOutput(ioutil.Discard)`.

If you think it would be a useful addition as a separate method, I'm happy to add it though.